src/main/java/HtmlUtil.java

```diff
@@ -10,41 +12,30 @@ public static String testableHtml(PageData pageData, boolean includeSuiteSetup)
 
         if (pageData.hasAttribute("Test")) {
             if (includeSuiteSetup) {
-                WikiPage suiteSetup = PageCrawlerImpl.getInheritedPage(SuiteResponder.SUITE_SETUP_NAME, wikiPage);
-                if (suiteSetup != null) {
-                    WikiPagePath pagePath = wikiPage.getPageCrawler().getFullPath(suiteSetup);
-                    String pagePathName = PathParser.render(pagePath);
-                    buffer.append("!include -setup .").append(pagePathName).append("\n");
-                }
-            }
-            WikiPage setup = PageCrawlerImpl.getInheritedPage("SetUp", wikiPage);
-            if (setup != null) {
-                WikiPagePath setupPath = wikiPage.getPageCrawler().getFullPath(setup);
-                String setupPathName = PathParser.render(setupPath);
-                buffer.append("!include -setup .").append(setupPathName).append("\n");
+                appendPathName(wikiPage, buffer, SuiteResponder.SUITE_SETUP_NAME, "!include -setup .");
             }
+            appendPathName(wikiPage, buffer, "SetUp", "!include -setup .");
         }
 
         buffer.append(pageData.getContent());
+        //TODO remove duplicated if
         if (pageData.hasAttribute("Test")) {
-            WikiPage teardown = PageCrawlerImpl.getInheritedPage("TearDown", wikiPage);
-            if (teardown != null) {
-                WikiPagePath tearDownPath = wikiPage.getPageCrawler().getFullPath(teardown);
-                String tearDownPathName = PathParser.render(tearDownPath);
-                buffer.append("!include -teardown .").append(tearDownPathName).append("\n");
-            }
+            appendPathName(wikiPage, buffer, "TearDown", "!include -teardown .");
             if (includeSuiteSetup) {
-                WikiPage suiteTeardown = PageCrawlerImpl.getInheritedPage(SuiteResponder.SUITE_TEARDOWN_NAME, wikiPage);
-                if (suiteTeardown != null) {
-                    WikiPagePath pagePath = wikiPage.getPageCrawler().getFullPath(suiteTeardown);
-                    String pagePathName = PathParser.render(pagePath);
-                    buffer.append("!include -teardown .").append(pagePathName).append("\n");
-                }
+                appendPathName(wikiPage, buffer, SuiteResponder.SUITE_TEARDOWN_NAME, "!include -teardown .");
             }
         }
 
         pageData.setContent(buffer.toString());
         return pageData.getHtml();
     }
 
+    private static void appendPathName(WikiPage wikiPage, StringBuffer buffer, String pageName, String appendKey) throws Exception {
```

Comment:
Think about handling this exception or throw a more precise one

Reply:
Context information is missing in function name. E.g.: appendPathNameOfPage. appendKey parameter is unintuitive without looking into actual function body.